docs/release-process.md

```diff
@@ -1,9 +1,17 @@
-# Release Process for NGINX Kubernetes Gateway
+# Releases
 
-NGINX Kubernetes Gateway uses semantic versioning for its releases. For more information see https://semver.org.
+NGINX Kubernetes Gateway (NKG) uses semantic versioning for its releases. For more information see https://semver.org.
 
 Warning: Major version zero (0.y.z) is reserved for development, anything MAY change at any time. The public API is not stable.
 
+NKG, especially in its early stages, will release periodically on a feature set cadence. By feature set, we intend to mean 2 or 3 features as tracked by the repository's Github issues. We do this to remain agile and to expose features relatively quickly, but also to minimize the cost of many frequent releases and the associated process overhead.
+
+This means users can expect a release generally every 1 or 2 months, and can also track the progress of a release and its features by viewing our Github issues. Each issue scheduled for a release will be labelled with "proposal" and its scheduled release, the label will use the form vMAJOR.MINOR. Multiple issues may be written to describe one use-case and each will be labelled according to this process. Sub-issues should also link to their parent (the parent issue will list the links via mention notifications).
```

Comment:
for releases, GitHub milestones will be more appropriate rather than a label.  https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/about-milestones

Reply:
I agree that milestones work well for this, we can name the milestone with the tag we expect to release and apply it to the issue.

(see https://github.com/nginxinc/nginx-prometheus-exporter/milestone/4?closed=1 for an example)

Reply:
Great, never used milestones so was unfamiliar. Updating.